ldap_probe/models.py

```diff
@@ -187,6 +219,308 @@ def get_node(self):  # pylint: disable=inconsistent-return-statements
+            this argument is provided
+
+            If this argument is `None`, this method will not filter for
+            performance degradation results.
+
+            Otherwise, the argument will be updated to a
+            :class:`deciaml.Decimal` value based on the original value:
+
+            * if the value is 'warning', update the argument to the value
+              provided by the user preference defined in
+              :class:`citrus_borg.dynamic_preferences_registry.LdapPerfWarnTreshold`
+
+            * else if the value is 'alert', update the argument to the value
+              provided by the user preference defined in
+              :class:`citrus_borg.dynamic_preferences_registry.LdapPerfAlertTreshold`
+
+            * else, try to convert the original value to
+              :class:`deciaml.Decimal` and raise a
+              :exc:`exceptions.ValueError` if the conversion fails
+
+        :arg time_delta_args: optional named arguments that are used to
+            initialize a :class:`datetime.duration` object
+
+            If not present, the method will use the period defined by the
+            :class:`citrus_borg.dynamic_preferences_registry.LdapReportPeriod`
+            user preference
+
+        :returns:
+
+            * the moments used to filter the data in the report by time
+
+            * the :attr:`subscription
+              <ssl_cert_tracker.models.Subscription.subscription>` that will
+              be used to deliver this report via email
+
+            * the :class:`django.db.models.query.QuerySet` based on one
+              of the classes inheriting from  the :class:`BaseADNode` abstract
+              model
+
+              Depending on the calling class the following aggregates with
+              regards to LDAP probes are placed in the queryset
+
+                  * the number of failed probes
+
+                  * the number of successful probes
+
+                  * average timing for initialize calls of the successful
+                    probes
+
+                  * min, max, average timing for `bind` calls or `anonymous
+                   bind` calls of the successful probes
+
+                  * min, max, average timing for `extended search` calls or
+                    `read root dse` calls of the successful probes
+
+        :raises:
+
+            :exc:`exceptions.ValueError` if the `perf_filter` argument
+            is not `None` and it cannot be used to initialize a
+            :class:`decimal.Decimal` object
+
+        .. todo::
+
+            Add a catch for no nodes available.
+
+            This is a bad thing if there
+            are no orion nodes (treat this case as a fully separate alert).
+
+            However, the normal case is that all `AD` nodes are now
+            defined in Orion and in that case, we can safely abort
+            report calls for non Orion nodes.
+
+
+        """
+        def resolve_perf_filter(perf_filter):
```

Comment:
Why not just run this code, since it is only used once and right after the definition?

Reply:
just cutting down in the number of branches in the method. it is already a fat method and pylint tends to start bitching if you do a lot of branching in the same method